content/api/loaders.md

```diff
@@ -3,4 +3,302 @@ title: Loader API
+
+### `loaders`
+
+``` javascript
+loaders = [{request: string, path: string, query: string, module: function}]
+```
+
+An array of all the loaders. It is writeable in the pitch phase.
+
+In the example:
+
+``` javascript
+[
+  { request: "/abc/loader1.js?xyz",
+	path: "/abc/loader1.js",
+	query: "?xyz",
+	module: [Function]
+  },
+  { request: "/abc/node_modules/loader2/index.js",
+	path: "/abc/node_modules/loader2/index.js",
+	query: "",
+	module: [Function]
+  }
+]
+```
+
+### `loaderIndex`
+
+The index in the loaders array of the current loader.
+
+In the example: in loader1: `0`, in loader2: `1`
+
+### `resource`
+
+The resource part of the request, including query.
+
+In the example: `"/abc/resource.js?rrr"`
+
+### `resourcePath`
+
+The resource file.
+
+In the example: `"/abc/resource.js"`
+
+### `resourceQuery`
+
+The query of the resource.
+
+In the example: `"?rrr"`
+
+### `emitWarning`
+
+``` javascript
+emitWarning(message: string)
+```
+
+Emit a warning.
+
+### `emitError`
+
+``` javascript
+emitError(message: string)
+```
+
+Emit an error.
+
+### `exec`
+
+``` javascript
+exec(code: string, filename: string)
+```
+
+Execute some code fragment like a module.
+
+T> Don't use `require(this.resourcePath)`, use this function to make loaders chainable!
+
+### `resolve`
+
+``` javascript
+resolve(context: string, request: string, callback: function(err, result: string))
+```
+
+Resolve a request like a require expression.
+
+### `resolveSync`
+
+``` javascript
+resolveSync(context: string, request: string) -> string
+```
+
+Resolve a request like a require expression.
+
+### `addDependency`
+
+``` javascript
+addDependency(file: string)
+dependency(file: string) // shortcut
+```
+
+Add a file as dependency of the loader result in order to make them watchable.
+
+### `addContextDependency`
+
+``` javascript
+addContextDependency(directory: string)
+```
+
+Add a directory as dependency of the loader result.
+
+### `clearDependencies`
+
+``` javascript
+clearDependencies()
+```
+
+Remove all dependencies of the loader result. Even initial dependencies and these of other loaders. Consider using `pitch`.
+
+### `value`
+
+Pass values to the next loader. If you know what your result exports if executed as module, set this value here (as a only element array).
+
+### `inputValue`
+
+Passed from the last loader. If you would execute the input argument as module, consider reading this variable for a shortcut (for performance).
+
+### `options`
+
+The options passed to the Compiler.
+
+### `debug`
+
+A boolean flag. It is set when in debug mode.
+
+### `minimize`
+
+Should the result be minimized.
+
+### `sourceMap`
+
+Should a SourceMap be generated.
+
+### `target`
+
+Target of compilation. Passed from configuration options.
+
+Example values: `"web"`, `"node"`
+
+### `webpack`
+
+Set to true when this is compiled by webpack.
```

Comment:
Loader runner is a separate module so could be used externally. I should mark that.